### PR TITLE
[GUVNOR-3481] make incremental build messages to include KieBaseName

### PR DESCRIPTION
**Pull request by mariofusco**

### CODE REVIEW

File: drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/KieBuilderSetImpl.java [hunk 1]
```diff
@@ -154,15 +153,25 @@ private IncrementalResults buildChanges(Collection<String> filesToBuild) {
...
     }
 
-    private IncrementalResultsImpl getIncrementalResults(List<KnowledgeBuilderError> currentErrors) {
+    private IncrementalResultsImpl getIncrementalResults(Map<String, Collection<KnowledgeBuilderError>> currentErrors) {
         IncrementalResultsImpl results = new IncrementalResultsImpl();
-        for (KnowledgeBuilderError error : currentErrors) {
-            if (!previousErrors.remove(error)) {
-                results.addMessage(error);
+        for (Map.Entry<String, Collection<KnowledgeBuilderError>> entry : currentErrors.entrySet()) {
+            Collection<KnowledgeBuilderError> previousErrorsInKB = previousErrors.remove(entry.getValue());
```
Comment (manstis) [inline drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/KieBuilderSetImpl.java]: Should this be ```previousErrors.remove(entry.getKey())```?

**Review by manstis — COMMENTED** [review]
@mariofusco One question.. which might explain the failing unit tests for this PR?

File: kie-ci/src/test/java/org/kie/scanner/KieModuleIncrementalCompilationTest.java [hunk 2]
```diff
@@ -212,4 +216,54 @@ public void checkIncrementalCompilationWithFunctionThenRule() throws Exception {
...
+                  rule);
+        KieBuilder kieBuilder = ks.newKieBuilder(kfs).buildAll();
+        assertEquals(0,
+                     kieBuilder.getResults().getMessages().size());
+
+        kfs.write("src/main/resources/org/kie/scanner/invalidRule.drl",
+                  invalidRule);
+        IncrementalResults addResults = ((InternalKieBuilder) kieBuilder).createFileSet("src/main/resources/org/kie/scanner/invalidRule.drl").build();
+
+        assertFalse(addResults.getAddedMessages().isEmpty());
```
Comment (jomarko) [inline kie-ci/src/test/java/org/kie/scanner/KieModuleIncrementalCompilationTest.java]: Would be possible here also count added messages please? We should be sure that count of issues equals to count of KBases in this simple case. I think.

File: drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/KieBuilderSetImpl.java (hunk 1 above)
Comment (jomarko) [reply drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/KieBuilderSetImpl.java]: Same question here as Mike.

**Review by manstis — APPROVED** [review]

**Review by jomarko — APPROVED** [review]
@mariofusco thanks for the updates.

**Comment by manstis** [comment]
@mariofusco Hi (and thanks) Mario. I need this on 6.5.x too. Could you please cherry-pick?

**Comment by manstis** [comment]
@mariofusco Sorry.. and 7.3.x :-(

**Comment by mariofusco** [comment]
@manstis I cherry-picked this improvement to both 7.3.x and 6.5.x branches.

**Pull request merged**